File: exercicios/time_conversion.py
import math
import os
import random
import re
import sys

def timeConversion(s):
    temp = ""
    ampmid = s[-2]
    daynight = s[0]+s[1]
    makeint = int(s[0])*10 + int(s[1])
    if ampmid == "A" and daynight == "12":
        temp += "00"
        temp += s[2:]
        return temp[:-2]
    elif ampmid == "P" and daynight != "12":
        a = makeint + 12
        b = str(a)
        temp += b
        temp += s[2:]
        return temp[:-2]
    else:
        temp = s
        return temp[:-2]   
   
tempo = "11:59:59PM"
timeConversion(tempo)

# link do exercício
# https://www.hackerrank.com/challenges/one-month-preparation-kit-time-conversion/problem?isFullScreen=true&h_l=interview&playlist_slugs%5B%5D=preparation-kits&playlist_slugs%5B%5D=one-month-preparation-kit&playlist_slugs%5B%5D=one-month-week-one

#Material de apoio
# https://note.nkmk.me/en/python-str-remove-strip/

File: exercicios/time_conversion.js
function timeConversion(s) {
    // Write your code here
    let temp = ""
    const ampmid = s.slice(-2)
    const daynight = s.slice(-10,-8)
    let makeint = parseInt(s.slice(-10,-9))*10 + parseInt(s.slice(-9,-8))
    if (ampmid == "AM" && daynight == "12") {
        temp += "00"
        temp += s.slice(2)
        return temp.slice(0,-2)
    } else if (ampmid == "PM" && daynight != "12") {
        let a = makeint + 12
        let b = a.toString()
        temp += b
        temp += s.slice(2)
        return temp.slice(0,-2)
    } else {
        temp = s
        return temp.slice(0,-2)
    }
}

const tempo = "11:59:59PM"
timeConversion(tempo)